{
 "cells": [
  {
   "cell_type": "markdown",
   "source": [
    "# VEuPathDB Data Report\n",
    "Reports for the data fields in the API. \n"
   ],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "source": [
    "import requests\n",
    "from IPython.display import display \n",
    "import requests \n",
    "import json\n",
    "import pandas as pd\n",
    "pd.set_option('display.max_rows', None)\n",
    "pd.set_option('display.max_colwidth', None)\n",
    "\n",
    " "
   ],
   "outputs": [],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "source": [
    "\n",
    "\"\"\"\n",
    "Helper Functions \n",
    "\"\"\"\n",
    "def http_request(req):\n",
    "    print(\"[INFO] status code response:\",req.status_code)\n",
    "    print(\"[INFO] reason:\",req.reason)\n",
    "    pretty_json=json.loads(req.text)\n",
    "    print(\"[INFO] text response:\", json.dumps(pretty_json, indent=4))\n"
   ],
   "outputs": [],
   "metadata": {}
  },
  {
   "cell_type": "markdown",
   "source": [
    "Run the request below to gather the datasets. "
   ],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "source": [
    "# all datasets Search\n",
    "all_datasets='https://veupathdb.org/veupathdb/service/record-types/dataset/searches/AllDatasets/reports/standard?reportConfig={\"attributes\":[\"primary_key\",\"organism_prefix\",\"project_id\",\"eupath_release\",\"newcategory\",\"summary\",\"contact\",\"wdk_weight\",\"version\",\"institution\",\"build_number_introduced\",\"pmids_download\",\"release_policy\",\"short_attribution\",\"type\",\"genecount\"],\"tables\":[\"Publications\",\"Contacts\",\"GenomeHistory\",\"DatasetHistory\",\"Version\",\"References\",\"HyperLinks\",\"GeneTypeCounts\",\"TranscriptTypeCounts\"],\"attributeFormat\":\"text\"}'\n",
    "\n",
    "all_req=requests.get(all_datasets)\n",
    "datasets_dict=all_req.json()\n"
   ],
   "outputs": [],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "source": [
    "records_list=datasets_dict['records']"
   ],
   "outputs": [],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "source": [
    "# example record\n",
    "#print(json.dumps(records_list[0], indent=4))"
   ],
   "outputs": [],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "source": [
    "def setup_df(_data, key, verbose=True):\n",
    "    _df = pd.DataFrame(_data, index=[0])\n",
    "    #_df.columns = [\"key\", \"value\"]\n",
    "\n",
    "    if key ==  'tables.GenomeHistory': tables_gh.append(_df)\n",
    "    elif key ==  'tables.Publications': tables_pubs.append(_df)\n",
    "    elif key ==  'tables.Contacts': tables_contacts.append(_df)\n",
    "    elif key ==  'tables.Version': tables_version.append(_df)\n",
    "    elif key ==  'tables.GeneTypeCounts': tables_gtc.append(_df)\n",
    "    elif key ==  'tables.References': tables_ref.append(_df)\n",
    "    elif key ==  'tables.DatasetHistory': tables_dh.append(_df)\n",
    "    elif key ==  'tables.HyperLinks': tables_hl.append(_df)\n",
    "    elif key ==  'tables.TranscriptTypeCounts': tables_ttc.append(_df)\n",
    "    \n",
    "    \n",
    "    if verbose == True:\n",
    "        display(_df)\n",
    "    return _df\n",
    "\n",
    "\n",
    "\n",
    "tables_gh=[]\n",
    "tables_pubs=[]\n",
    "tables_contacts=[]\n",
    "tables_version=[]\n",
    "tables_gtc=[]\n",
    "tables_ref=[]\n",
    "tables_dh=[]\n",
    "tables_hl=[]\n",
    "tables_ttc=[]\n",
    "\n",
    "attributes=[]\n",
    "\n",
    "for _dict in records_list:\n",
    "\n",
    "    _id=_dict['id'][0]['value']\n",
    "    #print('[INFO] dataset id: ', _id)\n",
    "    verbose=False\n",
    "    if verbose==True:\n",
    "        print(\"[INFO] keys: \", _dict.keys())\n",
    "        print(\"[INFO] displayName: \", _dict['displayName'])\n",
    "        print(\"[INFO] recordClassName: \", _dict['recordClassName'])\n",
    "        print(\"[INFO] id: \", _dict['id'])\n",
    "        print(\"[INFO] tableErrors: \", _dict['tableErrors'])\n",
    "\n",
    "    attributes_df=pd.DataFrame(_dict['attributes'], index=[0])\n",
    "    attributes.append(attributes_df)\n",
    "\n",
    "    ############################    \n",
    "    #  get available tables \n",
    "    ############################\n",
    "    for key in list(_dict['tables'].keys()):\n",
    "        # assign variables\n",
    "        _key=\"tables.\"+key\n",
    "        #print(\"elif key ==  '%s': tables_.append(_df)\"%_key )\n",
    "        temp_list=_dict['tables'][key]\n",
    "\n",
    "        \n",
    "        if len(temp_list) == 1:\n",
    "            _data=temp_list[0]\n",
    "            if type(_data) is dict: \n",
    "                setup_df(_data, _key, verbose=False)\n",
    "            else:\n",
    "                print(_data)\n",
    "\n",
    "        elif len(temp_list) > 1:\n",
    "            for _data in temp_list:\n",
    "                if type(_data) is dict:\n",
    "                    setup_df(_data, _key, verbose=False)\n",
    "                else:\n",
    "                    print(_data)\n",
    "\n",
    "        elif len(temp_list) == 0:\n",
    "            #print(\"[INFO] key: %s | val: %s \"%(key, temp_list))\n",
    "            pass\n",
    "\n",
    "  \n",
    "attr_df=pd.concat(attributes)\n",
    "#display(attr_df.head())\n",
    "\n",
    "df_gh=pd.concat(tables_gh)\n",
    "#display(df_gh.tail())\n",
    "\n",
    "df_pubs=pd.concat(tables_pubs)\n",
    "#display(df_pubs.tail())\n",
    "\n",
    "df_contacts=pd.concat(tables_contacts)\n",
    "#display(df_contacts.tail())\n",
    "\n",
    "df_version=pd.concat(tables_version)\n",
    "#display(df_version.tail())\n",
    "\n",
    "df_gtc=pd.concat(tables_gtc)\n",
    "#display(df_gtc.tail())\n",
    "\n",
    "df_ref=pd.concat(tables_ref)\n",
    "#display(df_ref.tail())\n",
    "\n",
    "df_dh=pd.concat(tables_dh)\n",
    "#display(df_dh.tail())\n",
    "\n",
    "df_hl=pd.concat(tables_hl)\n",
    "#display(df_hl.tail())\n",
    "\n",
    "df_ttc=pd.concat(tables_ttc)\n",
    "#display(df_ttc.tail())\n",
    "\n",
    "\n"
   ],
   "outputs": [],
   "metadata": {}
  },
  {
   "cell_type": "markdown",
   "source": [
    "## `records.attributes`"
   ],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "source": [
    "attr_df.head()"
   ],
   "outputs": [
    {
     "output_type": "execute_result",
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>summary</th>\n",
       "      <th>wdk_weight</th>\n",
       "      <th>build_number_introduced</th>\n",
       "      <th>organism_prefix</th>\n",
       "      <th>type</th>\n",
       "      <th>primary_key</th>\n",
       "      <th>version</th>\n",
       "      <th>release_policy</th>\n",
       "      <th>newcategory</th>\n",
       "      <th>pmids_download</th>\n",
       "      <th>institution</th>\n",
       "      <th>project_id</th>\n",
       "      <th>short_attribution</th>\n",
       "      <th>contact</th>\n",
       "      <th>genecount</th>\n",
       "      <th>eupath_release</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Genome Sequence for &lt;i&gt;Epichloe glyceriae&lt;/i&gt; E277</td>\n",
       "      <td>0</td>\n",
       "      <td>51</td>\n",
       "      <td>&lt;i&gt;Epichloe glyceriae&lt;/i&gt; E277&lt;br&gt;</td>\n",
       "      <td>genome</td>\n",
       "      <td>&lt;i&gt;Epichloe glyceriae&lt;/i&gt; E277 Genome Sequence</td>\n",
       "      <td>2011-09-02</td>\n",
       "      <td>None</td>\n",
       "      <td>Genomes</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/23468653</td>\n",
       "      <td>Department of Plant Pathology, University of Kentucky, USA</td>\n",
       "      <td>FungiDB</td>\n",
       "      <td>Schardl et al. 2013</td>\n",
       "      <td>Christopher Schardl</td>\n",
       "      <td>None</td>\n",
       "      <td>EuPathDB rel. 51, 2021-MAR-16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>DNA damage-induced transcriptome changes in Candida glabrata</td>\n",
       "      <td>0</td>\n",
       "      <td>56</td>\n",
       "      <td>&lt;i&gt;Candida glabrata&lt;/i&gt; CBS 138&lt;br&gt;</td>\n",
       "      <td>transcript_expression</td>\n",
       "      <td>DNA damage-induced transcriptome changes in Candida glabrata</td>\n",
       "      <td>2020-12-15</td>\n",
       "      <td>None</td>\n",
       "      <td>RNASeq</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/33323516</td>\n",
       "      <td>Center for Discovery and Innovation, Hackensack Meridian Health</td>\n",
       "      <td>FungiDB</td>\n",
       "      <td>Shor et al. 2020</td>\n",
       "      <td>David Perlin</td>\n",
       "      <td>None</td>\n",
       "      <td>EuPathDB rel. 56, 2022-FEB-15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Automatic mosquito surveillance from 2017-06-21 to 2017-12-31 by BG-Counter smart traps owned by Collier Mosquito Control District. (MapVEu VBP0000343)</td>\n",
       "      <td>0</td>\n",
       "      <td>47</td>\n",
       "      <td>&lt;br&gt;</td>\n",
       "      <td>population_biology</td>\n",
       "      <td>Mosquito surveillance by smart traps owned by Collier Mosquito Control District. 2017</td>\n",
       "      <td>2018-10-17</td>\n",
       "      <td>None</td>\n",
       "      <td>Population Biology</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/</td>\n",
       "      <td>Collier Mosquito Control District</td>\n",
       "      <td>VectorBase</td>\n",
       "      <td>Keira Lucas</td>\n",
       "      <td>Keira Lucas</td>\n",
       "      <td>None</td>\n",
       "      <td>EuPathDB rel. 47, 2020-APR-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>characterization of genetic differences between insecticide resistant and susceptible Culex quinquefasciatus mosquitoes using targeted DNA sequencing. (MapVEu VBP0000454)</td>\n",
       "      <td>0</td>\n",
       "      <td>47</td>\n",
       "      <td>&lt;br&gt;</td>\n",
       "      <td>population_biology</td>\n",
       "      <td>Using targeted next-generation sequencing to characterize genetic differences associated with insecticide resistance in Culex quinquefasciatus populations from the southern U.S.</td>\n",
       "      <td>2019-05-22</td>\n",
       "      <td>None</td>\n",
       "      <td>Population Biology</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/</td>\n",
       "      <td>Division of Vector-Borne Diseases, Centers for Disease Control and Prevention, Fort Collins, Colorado, United States of America</td>\n",
       "      <td>VectorBase</td>\n",
       "      <td>Linda Kothera</td>\n",
       "      <td>Linda Kothera</td>\n",
       "      <td>None</td>\n",
       "      <td>EuPathDB rel. 47, 2020-APR-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>&lt;i&gt;G. lamblia&lt;/i&gt; Assemblage A Proteomics of virulent and avirulent Strains (Mass Spec peptides)</td>\n",
       "      <td>0</td>\n",
       "      <td>25</td>\n",
       "      <td>&lt;i&gt;Giardia Assemblage&lt;/i&gt; A isolate WB&lt;br&gt;</td>\n",
       "      <td>protein_expression</td>\n",
       "      <td>Virulent vs Avirulent Trophozoites (B-2041, H-106)</td>\n",
       "      <td>2014-11-12</td>\n",
       "      <td>None</td>\n",
       "      <td>Protein expression</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/25266764</td>\n",
       "      <td>Walter and Eliza Hall Institute of Medical Research</td>\n",
       "      <td>GiardiaDB</td>\n",
       "      <td>Emery et al.</td>\n",
       "      <td>Samantha J. Emery-Corbin</td>\n",
       "      <td>None</td>\n",
       "      <td>EuPathDB rel. 25, 2015-JUL-23</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                                                                                                                      summary  \\\n",
       "0                                                                                                                          Genome Sequence for <i>Epichloe glyceriae</i> E277   \n",
       "0                                                                                                                DNA damage-induced transcriptome changes in Candida glabrata   \n",
       "0                     Automatic mosquito surveillance from 2017-06-21 to 2017-12-31 by BG-Counter smart traps owned by Collier Mosquito Control District. (MapVEu VBP0000343)   \n",
       "0  characterization of genetic differences between insecticide resistant and susceptible Culex quinquefasciatus mosquitoes using targeted DNA sequencing. (MapVEu VBP0000454)   \n",
       "0                                                                            <i>G. lamblia</i> Assemblage A Proteomics of virulent and avirulent Strains (Mass Spec peptides)   \n",
       "\n",
       "  wdk_weight build_number_introduced  \\\n",
       "0          0                      51   \n",
       "0          0                      56   \n",
       "0          0                      47   \n",
       "0          0                      47   \n",
       "0          0                      25   \n",
       "\n",
       "                              organism_prefix                   type  \\\n",
       "0          <i>Epichloe glyceriae</i> E277<br>                 genome   \n",
       "0         <i>Candida glabrata</i> CBS 138<br>  transcript_expression   \n",
       "0                                        <br>     population_biology   \n",
       "0                                        <br>     population_biology   \n",
       "0  <i>Giardia Assemblage</i> A isolate WB<br>     protein_expression   \n",
       "\n",
       "                                                                                                                                                                         primary_key  \\\n",
       "0                                                                                                                                     <i>Epichloe glyceriae</i> E277 Genome Sequence   \n",
       "0                                                                                                                       DNA damage-induced transcriptome changes in Candida glabrata   \n",
       "0                                                                                              Mosquito surveillance by smart traps owned by Collier Mosquito Control District. 2017   \n",
       "0  Using targeted next-generation sequencing to characterize genetic differences associated with insecticide resistance in Culex quinquefasciatus populations from the southern U.S.   \n",
       "0                                                                                                                                 Virulent vs Avirulent Trophozoites (B-2041, H-106)   \n",
       "\n",
       "      version release_policy         newcategory  \\\n",
       "0  2011-09-02           None             Genomes   \n",
       "0  2020-12-15           None              RNASeq   \n",
       "0  2018-10-17           None  Population Biology   \n",
       "0  2019-05-22           None  Population Biology   \n",
       "0  2014-11-12           None  Protein expression   \n",
       "\n",
       "                                 pmids_download  \\\n",
       "0  https://www.ncbi.nlm.nih.gov/pubmed/23468653   \n",
       "0  https://www.ncbi.nlm.nih.gov/pubmed/33323516   \n",
       "0          https://www.ncbi.nlm.nih.gov/pubmed/   \n",
       "0          https://www.ncbi.nlm.nih.gov/pubmed/   \n",
       "0  https://www.ncbi.nlm.nih.gov/pubmed/25266764   \n",
       "\n",
       "                                                                                                                       institution  \\\n",
       "0                                                                       Department of Plant Pathology, University of Kentucky, USA   \n",
       "0                                                                  Center for Discovery and Innovation, Hackensack Meridian Health   \n",
       "0                                                                                                Collier Mosquito Control District   \n",
       "0  Division of Vector-Borne Diseases, Centers for Disease Control and Prevention, Fort Collins, Colorado, United States of America   \n",
       "0                                                                              Walter and Eliza Hall Institute of Medical Research   \n",
       "\n",
       "   project_id    short_attribution                   contact genecount  \\\n",
       "0     FungiDB  Schardl et al. 2013       Christopher Schardl      None   \n",
       "0     FungiDB     Shor et al. 2020              David Perlin      None   \n",
       "0  VectorBase          Keira Lucas               Keira Lucas      None   \n",
       "0  VectorBase        Linda Kothera             Linda Kothera      None   \n",
       "0   GiardiaDB         Emery et al.  Samantha J. Emery-Corbin      None   \n",
       "\n",
       "                  eupath_release  \n",
       "0  EuPathDB rel. 51, 2021-MAR-16  \n",
       "0  EuPathDB rel. 56, 2022-FEB-15  \n",
       "0  EuPathDB rel. 47, 2020-APR-07  \n",
       "0  EuPathDB rel. 47, 2020-APR-07  \n",
       "0  EuPathDB rel. 25, 2015-JUL-23  "
      ]
     },
     "metadata": {},
     "execution_count": 12
    }
   ],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "source": [
    "attr_df['organism_prefix'].head()"
   ],
   "outputs": [
    {
     "output_type": "execute_result",
     "data": {
      "text/plain": [
       "0            <i>Epichloe glyceriae</i> E277<br>\n",
       "0           <i>Candida glabrata</i> CBS 138<br>\n",
       "0                                          <br>\n",
       "0                                          <br>\n",
       "0    <i>Giardia Assemblage</i> A isolate WB<br>\n",
       "Name: organism_prefix, dtype: object"
      ]
     },
     "metadata": {},
     "execution_count": 15
    }
   ],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "source": [
    "attr_df[['type', 'newcategory']].tail(10)"
   ],
   "outputs": [
    {
     "output_type": "execute_result",
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>type</th>\n",
       "      <th>newcategory</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>transcript_expression</td>\n",
       "      <td>DNA Microarray Assay</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>genome</td>\n",
       "      <td>Genomes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>population_biology</td>\n",
       "      <td>Population Biology</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>transcript_expression</td>\n",
       "      <td>RNASeq</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>protein_expression</td>\n",
       "      <td>Protein expression</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>transcript_expression</td>\n",
       "      <td>RNASeq</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>population_biology</td>\n",
       "      <td>Population Biology</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>transcript_expression</td>\n",
       "      <td>RNASeq</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>isolates</td>\n",
       "      <td>Genetic variation</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>genome</td>\n",
       "      <td>Genomes</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                    type           newcategory\n",
       "0  transcript_expression  DNA Microarray Assay\n",
       "0                 genome               Genomes\n",
       "0     population_biology    Population Biology\n",
       "0  transcript_expression                RNASeq\n",
       "0     protein_expression    Protein expression\n",
       "0  transcript_expression                RNASeq\n",
       "0     population_biology    Population Biology\n",
       "0  transcript_expression                RNASeq\n",
       "0               isolates     Genetic variation\n",
       "0                 genome               Genomes"
      ]
     },
     "metadata": {},
     "execution_count": 9
    }
   ],
   "metadata": {}
  },
  {
   "cell_type": "markdown",
   "source": [
    "## `records.tables`"
   ],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "source": [
    "display(df_version['organism'].head())"
   ],
   "outputs": [
    {
     "output_type": "display_data",
     "data": {
      "text/plain": [
       "0            Epichloe glyceriae E277\n",
       "0           Candida glabrata CBS 138\n",
       "0    Giardia Assemblage A isolate WB\n",
       "0            Aedes albopictus Foshan\n",
       "0             Aedes aegypti LVP_AGWG\n",
       "Name: organism, dtype: object"
      ]
     },
     "metadata": {}
    }
   ],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "source": [
    "import numpy as np"
   ],
   "outputs": [],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "source": [
    "display(df_contacts['affiliation'].head())\n",
    "display(attr_df['institution'].head())\n"
   ],
   "outputs": [
    {
     "output_type": "display_data",
     "data": {
      "text/plain": [
       "0                                                                         Department of Plant Pathology, University of Kentucky, USA\n",
       "0                                                                    Center for Discovery and Innovation, Hackensack Meridian Health\n",
       "0                                                                    Center for Discovery and Innovation, Hackensack Meridian Health\n",
       "0                                                                                                  Collier Mosquito Control District\n",
       "0    Division of Vector-Borne Diseases, Centers for Disease Control and Prevention, Fort Collins, Colorado, United States of America\n",
       "Name: affiliation, dtype: object"
      ]
     },
     "metadata": {}
    },
    {
     "output_type": "display_data",
     "data": {
      "text/plain": [
       "0                                                                         Department of Plant Pathology, University of Kentucky, USA\n",
       "0                                                                    Center for Discovery and Innovation, Hackensack Meridian Health\n",
       "0                                                                                                  Collier Mosquito Control District\n",
       "0    Division of Vector-Borne Diseases, Centers for Disease Control and Prevention, Fort Collins, Colorado, United States of America\n",
       "0                                                                                Walter and Eliza Hall Institute of Medical Research\n",
       "Name: institution, dtype: object"
      ]
     },
     "metadata": {}
    },
    {
     "output_type": "execute_result",
     "data": {
      "text/plain": [
       "False"
      ]
     },
     "metadata": {},
     "execution_count": 20
    }
   ],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "source": [
    " df_gtc.head()"
   ],
   "outputs": [
    {
     "output_type": "execute_result",
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_count</th>\n",
       "      <th>gene_type</th>\n",
       "      <th>dataset_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>337</td>\n",
       "      <td>ncRNA gene</td>\n",
       "      <td>DS_01b2dc5b05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>4886</td>\n",
       "      <td>protein coding gene</td>\n",
       "      <td>DS_01b2dc5b05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>9</td>\n",
       "      <td>pseudogene</td>\n",
       "      <td>DS_01b2dc5b05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>173</td>\n",
       "      <td>ncRNA gene</td>\n",
       "      <td>DS_01c6cee55d</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>9869</td>\n",
       "      <td>protein coding gene</td>\n",
       "      <td>DS_01c6cee55d</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  gene_count            gene_type     dataset_id\n",
       "0        337           ncRNA gene  DS_01b2dc5b05\n",
       "0       4886  protein coding gene  DS_01b2dc5b05\n",
       "0          9           pseudogene  DS_01b2dc5b05\n",
       "0        173           ncRNA gene  DS_01c6cee55d\n",
       "0       9869  protein coding gene  DS_01c6cee55d"
      ]
     },
     "metadata": {},
     "execution_count": 11
    }
   ],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "source": [
    "attr_df.tail()"
   ],
   "outputs": [
    {
     "output_type": "execute_result",
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>summary</th>\n",
       "      <th>wdk_weight</th>\n",
       "      <th>build_number_introduced</th>\n",
       "      <th>organism_prefix</th>\n",
       "      <th>type</th>\n",
       "      <th>primary_key</th>\n",
       "      <th>version</th>\n",
       "      <th>release_policy</th>\n",
       "      <th>newcategory</th>\n",
       "      <th>pmids_download</th>\n",
       "      <th>institution</th>\n",
       "      <th>project_id</th>\n",
       "      <th>short_attribution</th>\n",
       "      <th>contact</th>\n",
       "      <th>genecount</th>\n",
       "      <th>eupath_release</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Mouse transcriptome time course during infecti...</td>\n",
       "      <td>0</td>\n",
       "      <td>53</td>\n",
       "      <td>&lt;i&gt;Crithidia fasciculata&lt;/i&gt; strain Cf-Cl&lt;br&gt;&lt;...</td>\n",
       "      <td>transcript_expression</td>\n",
       "      <td>Mouse transcriptome during P. vinckei CY infec...</td>\n",
       "      <td>2019-01-25</td>\n",
       "      <td>None</td>\n",
       "      <td>RNASeq</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/30683099</td>\n",
       "      <td>King Abdullah University of Science and Techno...</td>\n",
       "      <td>HostDB</td>\n",
       "      <td>Ramaprasad et al</td>\n",
       "      <td>Arnab Pain</td>\n",
       "      <td>None</td>\n",
       "      <td>EuPathDB rel. 53, 2021-JUL-21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>In 2011, Benin&amp;#8217;s National Malaria Contro...</td>\n",
       "      <td>0</td>\n",
       "      <td>47</td>\n",
       "      <td>&lt;br&gt;</td>\n",
       "      <td>population_biology</td>\n",
       "      <td>Resistance status of A. gambiae s.l. following...</td>\n",
       "      <td>2020-10-15</td>\n",
       "      <td>None</td>\n",
       "      <td>Population Biology</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/31941494</td>\n",
       "      <td>Centre de Recherche Entomologique de Cotonou (...</td>\n",
       "      <td>VectorBase</td>\n",
       "      <td>Arthur Sovi</td>\n",
       "      <td>Arthur Sovi</td>\n",
       "      <td>None</td>\n",
       "      <td>EuPathDB rel. 47, 2020-APR-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Transcriptome of &lt;i&gt;T. cruzi&lt;/i&gt; Y strain duri...</td>\n",
       "      <td>0</td>\n",
       "      <td>34</td>\n",
       "      <td>&lt;i&gt;Trypanosoma cruzi&lt;/i&gt; strain CL Brener&lt;br&gt;</td>\n",
       "      <td>transcript_expression</td>\n",
       "      <td>Transcriptome during Infection in Human Fibrob...</td>\n",
       "      <td>2016-04-05</td>\n",
       "      <td>None</td>\n",
       "      <td>RNASeq</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/27046031</td>\n",
       "      <td>Harvard T.H. Chan School of Public Health</td>\n",
       "      <td>TriTrypDB</td>\n",
       "      <td>Li et al.</td>\n",
       "      <td>Barbara Burleigh</td>\n",
       "      <td>None</td>\n",
       "      <td>EuPathDB rel. 34, 2017-SEP-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Whole-genome sequencing (WGS) was performed fo...</td>\n",
       "      <td>0</td>\n",
       "      <td>38</td>\n",
       "      <td>&lt;i&gt;Aspergillus fumigatus&lt;/i&gt; Af293&lt;br&gt;</td>\n",
       "      <td>isolates</td>\n",
       "      <td>Genomic Context of Azole-Resistance Mutations ...</td>\n",
       "      <td>2015-04-24</td>\n",
       "      <td>None</td>\n",
       "      <td>Genetic variation</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/26037120</td>\n",
       "      <td>Imperial College London</td>\n",
       "      <td>FungiDB</td>\n",
       "      <td>Matthew Fisher</td>\n",
       "      <td>Matthew Fisher</td>\n",
       "      <td>None</td>\n",
       "      <td>EuPathDB rel. 38, 2018-JUL-05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Genome Sequence and Annotation of &lt;i&gt;Hansenias...</td>\n",
       "      <td>0</td>\n",
       "      <td>52</td>\n",
       "      <td>&lt;i&gt;Hanseniaspora guilliermondii&lt;/i&gt; strain UTA...</td>\n",
       "      <td>genome</td>\n",
       "      <td>&lt;i&gt;Hanseniaspora guilliermondii&lt;/i&gt; strain UTA...</td>\n",
       "      <td>2017-02-02</td>\n",
       "      <td>None</td>\n",
       "      <td>Genomes</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pubmed/30462193</td>\n",
       "      <td>Department of Bioengineering, Institute of Bio...</td>\n",
       "      <td>FungiDB</td>\n",
       "      <td>Nuno Mira</td>\n",
       "      <td>Nuno Mira</td>\n",
       "      <td>4152</td>\n",
       "      <td>EuPathDB rel. 52, 2021-MAY-20</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                             summary wdk_weight  \\\n",
       "0  Mouse transcriptome time course during infecti...          0   \n",
       "0  In 2011, Benin&#8217;s National Malaria Contro...          0   \n",
       "0  Transcriptome of <i>T. cruzi</i> Y strain duri...          0   \n",
       "0  Whole-genome sequencing (WGS) was performed fo...          0   \n",
       "0  Genome Sequence and Annotation of <i>Hansenias...          0   \n",
       "\n",
       "  build_number_introduced                                    organism_prefix  \\\n",
       "0                      53  <i>Crithidia fasciculata</i> strain Cf-Cl<br><...   \n",
       "0                      47                                               <br>   \n",
       "0                      34      <i>Trypanosoma cruzi</i> strain CL Brener<br>   \n",
       "0                      38             <i>Aspergillus fumigatus</i> Af293<br>   \n",
       "0                      52  <i>Hanseniaspora guilliermondii</i> strain UTA...   \n",
       "\n",
       "                    type                                        primary_key  \\\n",
       "0  transcript_expression  Mouse transcriptome during P. vinckei CY infec...   \n",
       "0     population_biology  Resistance status of A. gambiae s.l. following...   \n",
       "0  transcript_expression  Transcriptome during Infection in Human Fibrob...   \n",
       "0               isolates  Genomic Context of Azole-Resistance Mutations ...   \n",
       "0                 genome  <i>Hanseniaspora guilliermondii</i> strain UTA...   \n",
       "\n",
       "      version release_policy         newcategory  \\\n",
       "0  2019-01-25           None              RNASeq   \n",
       "0  2020-10-15           None  Population Biology   \n",
       "0  2016-04-05           None              RNASeq   \n",
       "0  2015-04-24           None   Genetic variation   \n",
       "0  2017-02-02           None             Genomes   \n",
       "\n",
       "                                 pmids_download  \\\n",
       "0  https://www.ncbi.nlm.nih.gov/pubmed/30683099   \n",
       "0  https://www.ncbi.nlm.nih.gov/pubmed/31941494   \n",
       "0  https://www.ncbi.nlm.nih.gov/pubmed/27046031   \n",
       "0  https://www.ncbi.nlm.nih.gov/pubmed/26037120   \n",
       "0  https://www.ncbi.nlm.nih.gov/pubmed/30462193   \n",
       "\n",
       "                                         institution  project_id  \\\n",
       "0  King Abdullah University of Science and Techno...      HostDB   \n",
       "0  Centre de Recherche Entomologique de Cotonou (...  VectorBase   \n",
       "0          Harvard T.H. Chan School of Public Health   TriTrypDB   \n",
       "0                            Imperial College London     FungiDB   \n",
       "0  Department of Bioengineering, Institute of Bio...     FungiDB   \n",
       "\n",
       "  short_attribution           contact genecount                 eupath_release  \n",
       "0  Ramaprasad et al        Arnab Pain      None  EuPathDB rel. 53, 2021-JUL-21  \n",
       "0       Arthur Sovi       Arthur Sovi      None  EuPathDB rel. 47, 2020-APR-07  \n",
       "0         Li et al.  Barbara Burleigh      None  EuPathDB rel. 34, 2017-SEP-07  \n",
       "0    Matthew Fisher    Matthew Fisher      None  EuPathDB rel. 38, 2018-JUL-05  \n",
       "0         Nuno Mira         Nuno Mira      4152  EuPathDB rel. 52, 2021-MAY-20  "
      ]
     },
     "metadata": {},
     "execution_count": 12
    }
   ],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "source": [
    "df_gh_=df_gh.set_index(\"dataset_id\")"
   ],
   "outputs": [],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "source": [
    "df_gh_.head(50)"
   ],
   "outputs": [
    {
     "output_type": "execute_result",
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>note</th>\n",
       "      <th>annotation_version</th>\n",
       "      <th>annotation_source</th>\n",
       "      <th>release_number</th>\n",
       "      <th>build</th>\n",
       "      <th>release_date</th>\n",
       "      <th>genome_version</th>\n",
       "      <th>genome_source</th>\n",
       "      <th>functional_annotation_source</th>\n",
       "      <th>functional_annotation_version</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>dataset_id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>DS_0011425ac1</th>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>51</td>\n",
       "      <td>51</td>\n",
       "      <td>2021-03-16</td>\n",
       "      <td>GCA_000225285.2</td>\n",
       "      <td>INSDC</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DS_0024327063</th>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>56</td>\n",
       "      <td>56</td>\n",
       "      <td>2022-02-15</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DS_002a5dad83</th>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>47</td>\n",
       "      <td>47</td>\n",
       "      <td>2020-04-07</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DS_00672484d6</th>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>47</td>\n",
       "      <td>47</td>\n",
       "      <td>2020-04-07</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DS_006cb7d943</th>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>25</td>\n",
       "      <td>25</td>\n",
       "      <td>2015-07-23</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DS_00893e0bf8</th>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>47</td>\n",
       "      <td>47</td>\n",
       "      <td>2020-04-07</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DS_00b47a520a</th>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>47</td>\n",
       "      <td>47</td>\n",
       "      <td>2020-04-07</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DS_011c12b8d3</th>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>50</td>\n",
       "      <td>50</td>\n",
       "      <td>2020-12-17</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DS_01225327f4</th>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>48</td>\n",
       "      <td>48</td>\n",
       "      <td>2020-08-27</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DS_012298da2c</th>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>46</td>\n",
       "      <td>46</td>\n",
       "      <td>2019-11-06</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DS_019172d4ff</th>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>47</td>\n",
       "      <td>47</td>\n",
       "      <td>2020-04-07</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DS_01b2dc5b05</th>\n",
       "      <td>rebuild with INSDC accession added</td>\n",
       "      <td>Mar 23, 2015</td>\n",
       "      <td>GenBank</td>\n",
       "      <td>51</td>\n",
       "      <td>51</td>\n",
       "      <td>2021-03-16</td>\n",
       "      <td>GCA_000149845.2</td>\n",
       "      <td>GenBank</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DS_01b2dc5b05</th>\n",
       "      <td>None</td>\n",
       "      <td>5</td>\n",
       "      <td>Broad</td>\n",
       "      <td>2.19</td>\n",
       "      <td>19</td>\n",
       "      <td>2013-09-25</td>\n",
       "      <td>5</td>\n",
       "      <td>Broad</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DS_01b2dc5b05</th>\n",
       "      <td>re-annotated, data source changed</td>\n",
       "      <td>Mar 23, 2015</td>\n",
       "      <td>GenBank</td>\n",
       "      <td>29</td>\n",
       "      <td>29</td>\n",
       "      <td>2016-10-12</td>\n",
       "      <td>Mar 23, 2015</td>\n",
       "      <td>GenBank</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DS_01c6cee55d</th>\n",
       "      <td>newly-annotated</td>\n",
       "      <td>Nov 07, 2017</td>\n",
       "      <td>GenBank</td>\n",
       "      <td>46</td>\n",
       "      <td>46</td>\n",
       "      <td>2019-11-06</td>\n",
       "      <td>Nov 07, 2017</td>\n",
       "      <td>GenBank</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DS_01c6cee55d</th>\n",
       "      <td>consistent date format and INSDC accession whe...</td>\n",
       "      <td>Nov 07, 2017</td>\n",
       "      <td>GenBank</td>\n",
       "      <td>49</td>\n",
       "      <td>49</td>\n",
       "      <td>2020-11-05</td>\n",
       "      <td>GCA_000338675.2</td>\n",
       "      <td>GenBank</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DS_01c6cee55d</th>\n",
       "      <td>None</td>\n",
       "      <td>n/a</td>\n",
       "      <td>n/a</td>\n",
       "      <td>2.19</td>\n",
       "      <td>19</td>\n",
       "      <td>2013-09-25</td>\n",
       "      <td>Feb 7, 2013</td>\n",
       "      <td>GenBank</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DS_01d08f2525</th>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>47</td>\n",
       "      <td>47</td>\n",
       "      <td>2020-04-07</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DS_022384d18a</th>\n",
       "      <td>None</td>\n",
       "      <td>Nov 29, 2017</td>\n",
       "      <td>Sanger</td>\n",
       "      <td>36</td>\n",
       "      <td>36</td>\n",
       "      <td>2018-02-19</td>\n",
       "      <td>Nov 29, 2017</td>\n",
       "      <td>Sanger</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DS_022384d18a</th>\n",
       "      <td>rebuild with INSDC accession added</td>\n",
       "      <td>Nov 29, 2017</td>\n",
       "      <td>Sanger</td>\n",
       "      <td>51</td>\n",
       "      <td>51</td>\n",
       "      <td>2021-03-16</td>\n",
       "      <td>GCA_900097025.1</td>\n",
       "      <td>GenBank</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DS_0226b5025b</th>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>47</td>\n",
       "      <td>47</td>\n",
       "      <td>2020-04-07</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DS_02354d7d63</th>\n",
       "      <td>None</td>\n",
       "      <td>Sep 30, 2014</td>\n",
       "      <td>GenBank</td>\n",
       "      <td>24</td>\n",
       "      <td>24</td>\n",
       "      <td>2015-04-14</td>\n",
       "      <td>Sep 30, 2014</td>\n",
       "      <td>GenBank</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DS_02354d7d63</th>\n",
       "      <td>rebuild with INSDC accession added</td>\n",
       "      <td>Sep 30, 2014</td>\n",
       "      <td>GenBank</td>\n",
       "      <td>52</td>\n",
       "      <td>52</td>\n",
       "      <td>2021-05-20</td>\n",
       "      <td>GCA_000760515.1</td>\n",
       "      <td>GenBank</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DS_0241addc8e</th>\n",
       "      <td>None</td>\n",
       "      <td>Aug 27, 2021</td>\n",
       "      <td>RoditiLab</td>\n",
       "      <td>56</td>\n",
       "      <td>56</td>\n",
       "      <td>2022-02-15</td>\n",
       "      <td>GCA_019096175.1</td>\n",
       "      <td>INSDC</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DS_02717866d1</th>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>47</td>\n",
       "      <td>47</td>\n",
       "      <td>2020-04-07</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DS_0273077ad5</th>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>47</td>\n",
       "      <td>47</td>\n",
       "      <td>2020-04-07</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DS_02c0ef3cc2</th>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>48</td>\n",
       "      <td>48</td>\n",
       "      <td>2020-08-27</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DS_02caff887d</th>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>47</td>\n",
       "      <td>47</td>\n",
       "      <td>2020-04-07</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DS_03395fd8e3</th>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>47</td>\n",
       "      <td>47</td>\n",
       "      <td>2020-04-07</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DS_0339e44248</th>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>47</td>\n",
       "      <td>47</td>\n",
       "      <td>2020-04-07</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DS_033b09f437</th>\n",
       "      <td>functional annotation updated from CGD</td>\n",
       "      <td>Dec 11, 2013</td>\n",
       "      <td>CGD</td>\n",
       "      <td>53</td>\n",
       "      <td>53</td>\n",
       "      <td>2021-07-21</td>\n",
       "      <td>GCA_000149445.2</td>\n",
       "      <td>GenBank</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DS_033b09f437</th>\n",
       "      <td>rebuild with INSDC accession added</td>\n",
       "      <td>Dec 11, 2013</td>\n",
       "      <td>CGD</td>\n",
       "      <td>51</td>\n",
       "      <td>51</td>\n",
       "      <td>2021-03-16</td>\n",
       "      <td>GCA_000149445.2</td>\n",
       "      <td>GenBank</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DS_033b09f437</th>\n",
       "      <td>None</td>\n",
       "      <td>Dec 11, 2013</td>\n",
       "      <td>CGD</td>\n",
       "      <td>39</td>\n",
       "      <td>39</td>\n",
       "      <td>2018-08-30</td>\n",
       "      <td>Dec 11, 2013</td>\n",
       "      <td>CGD</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DS_033cf9c74d</th>\n",
       "      <td>None</td>\n",
       "      <td>Jun 12, 2013</td>\n",
       "      <td>GenBank</td>\n",
       "      <td>52</td>\n",
       "      <td>52</td>\n",
       "      <td>2021-05-20</td>\n",
       "      <td>GCF_000364345.1</td>\n",
       "      <td>INSDC</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DS_033e62f4ec</th>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>45</td>\n",
       "      <td>45</td>\n",
       "      <td>2019-09-05</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DS_035232648f</th>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>47</td>\n",
       "      <td>47</td>\n",
       "      <td>2020-04-07</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DS_0360a7ae66</th>\n",
       "      <td>None</td>\n",
       "      <td>Feb 27, 2015</td>\n",
       "      <td>GenBank</td>\n",
       "      <td>56</td>\n",
       "      <td>56</td>\n",
       "      <td>2022-02-15</td>\n",
       "      <td>GCA_000026945.1</td>\n",
       "      <td>INSDC</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DS_03824cf332</th>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>49</td>\n",
       "      <td>49</td>\n",
       "      <td>2020-11-05</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DS_03986f8069</th>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>47</td>\n",
       "      <td>47</td>\n",
       "      <td>2020-04-07</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DS_0399018270</th>\n",
       "      <td>rebuild with INSDC accession added</td>\n",
       "      <td>Jul 29, 2015</td>\n",
       "      <td>GenBank</td>\n",
       "      <td>51</td>\n",
       "      <td>51</td>\n",
       "      <td>2021-03-16</td>\n",
       "      <td>GCA_000182565.2</td>\n",
       "      <td>GenBank</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DS_0399018270</th>\n",
       "      <td>None</td>\n",
       "      <td>1</td>\n",
       "      <td>Broad</td>\n",
       "      <td>21</td>\n",
       "      <td>21</td>\n",
       "      <td>2014-05-08</td>\n",
       "      <td>1</td>\n",
       "      <td>Broad</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DS_0399018270</th>\n",
       "      <td>re-annotated, data source changed</td>\n",
       "      <td>Jul 29, 2015</td>\n",
       "      <td>GenBank</td>\n",
       "      <td>29</td>\n",
       "      <td>29</td>\n",
       "      <td>2016-10-12</td>\n",
       "      <td>Jul 29, 2015</td>\n",
       "      <td>GenBank</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DS_03a864a884</th>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>47</td>\n",
       "      <td>47</td>\n",
       "      <td>2020-04-07</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DS_03aef8200c</th>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>48</td>\n",
       "      <td>48</td>\n",
       "      <td>2020-08-27</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DS_03d8e25e02</th>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>47</td>\n",
       "      <td>47</td>\n",
       "      <td>2020-04-07</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DS_03f94e0879</th>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>47</td>\n",
       "      <td>47</td>\n",
       "      <td>2020-04-07</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DS_04102dc2b8</th>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>47</td>\n",
       "      <td>47</td>\n",
       "      <td>2020-04-07</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DS_0445efcab5</th>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>47</td>\n",
       "      <td>47</td>\n",
       "      <td>2020-04-07</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DS_04464ff60b</th>\n",
       "      <td>None</td>\n",
       "      <td>2018-11-27</td>\n",
       "      <td>GenBank</td>\n",
       "      <td>53</td>\n",
       "      <td>53</td>\n",
       "      <td>2021-07-21</td>\n",
       "      <td>GCA_003843895.1</td>\n",
       "      <td>INSDC</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DS_04666bdd43</th>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>28</td>\n",
       "      <td>28</td>\n",
       "      <td>2016-03-31</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                            note  \\\n",
       "dataset_id                                                         \n",
       "DS_0011425ac1                                               None   \n",
       "DS_0024327063                                               None   \n",
       "DS_002a5dad83                                               None   \n",
       "DS_00672484d6                                               None   \n",
       "DS_006cb7d943                                               None   \n",
       "DS_00893e0bf8                                               None   \n",
       "DS_00b47a520a                                               None   \n",
       "DS_011c12b8d3                                               None   \n",
       "DS_01225327f4                                               None   \n",
       "DS_012298da2c                                               None   \n",
       "DS_019172d4ff                                               None   \n",
       "DS_01b2dc5b05                 rebuild with INSDC accession added   \n",
       "DS_01b2dc5b05                                               None   \n",
       "DS_01b2dc5b05                  re-annotated, data source changed   \n",
       "DS_01c6cee55d                                    newly-annotated   \n",
       "DS_01c6cee55d  consistent date format and INSDC accession whe...   \n",
       "DS_01c6cee55d                                               None   \n",
       "DS_01d08f2525                                               None   \n",
       "DS_022384d18a                                               None   \n",
       "DS_022384d18a                 rebuild with INSDC accession added   \n",
       "DS_0226b5025b                                               None   \n",
       "DS_02354d7d63                                               None   \n",
       "DS_02354d7d63                 rebuild with INSDC accession added   \n",
       "DS_0241addc8e                                               None   \n",
       "DS_02717866d1                                               None   \n",
       "DS_0273077ad5                                               None   \n",
       "DS_02c0ef3cc2                                               None   \n",
       "DS_02caff887d                                               None   \n",
       "DS_03395fd8e3                                               None   \n",
       "DS_0339e44248                                               None   \n",
       "DS_033b09f437             functional annotation updated from CGD   \n",
       "DS_033b09f437                 rebuild with INSDC accession added   \n",
       "DS_033b09f437                                               None   \n",
       "DS_033cf9c74d                                               None   \n",
       "DS_033e62f4ec                                               None   \n",
       "DS_035232648f                                               None   \n",
       "DS_0360a7ae66                                               None   \n",
       "DS_03824cf332                                               None   \n",
       "DS_03986f8069                                               None   \n",
       "DS_0399018270                 rebuild with INSDC accession added   \n",
       "DS_0399018270                                               None   \n",
       "DS_0399018270                  re-annotated, data source changed   \n",
       "DS_03a864a884                                               None   \n",
       "DS_03aef8200c                                               None   \n",
       "DS_03d8e25e02                                               None   \n",
       "DS_03f94e0879                                               None   \n",
       "DS_04102dc2b8                                               None   \n",
       "DS_0445efcab5                                               None   \n",
       "DS_04464ff60b                                               None   \n",
       "DS_04666bdd43                                               None   \n",
       "\n",
       "              annotation_version annotation_source release_number build  \\\n",
       "dataset_id                                                                \n",
       "DS_0011425ac1               None              None             51    51   \n",
       "DS_0024327063               None              None             56    56   \n",
       "DS_002a5dad83               None              None             47    47   \n",
       "DS_00672484d6               None              None             47    47   \n",
       "DS_006cb7d943               None              None             25    25   \n",
       "DS_00893e0bf8               None              None             47    47   \n",
       "DS_00b47a520a               None              None             47    47   \n",
       "DS_011c12b8d3               None              None             50    50   \n",
       "DS_01225327f4               None              None             48    48   \n",
       "DS_012298da2c               None              None             46    46   \n",
       "DS_019172d4ff               None              None             47    47   \n",
       "DS_01b2dc5b05       Mar 23, 2015           GenBank             51    51   \n",
       "DS_01b2dc5b05                  5             Broad           2.19    19   \n",
       "DS_01b2dc5b05       Mar 23, 2015           GenBank             29    29   \n",
       "DS_01c6cee55d       Nov 07, 2017           GenBank             46    46   \n",
       "DS_01c6cee55d       Nov 07, 2017           GenBank             49    49   \n",
       "DS_01c6cee55d                n/a               n/a           2.19    19   \n",
       "DS_01d08f2525               None              None             47    47   \n",
       "DS_022384d18a       Nov 29, 2017            Sanger             36    36   \n",
       "DS_022384d18a       Nov 29, 2017            Sanger             51    51   \n",
       "DS_0226b5025b               None              None             47    47   \n",
       "DS_02354d7d63       Sep 30, 2014           GenBank             24    24   \n",
       "DS_02354d7d63       Sep 30, 2014           GenBank             52    52   \n",
       "DS_0241addc8e       Aug 27, 2021         RoditiLab             56    56   \n",
       "DS_02717866d1               None              None             47    47   \n",
       "DS_0273077ad5               None              None             47    47   \n",
       "DS_02c0ef3cc2               None              None             48    48   \n",
       "DS_02caff887d               None              None             47    47   \n",
       "DS_03395fd8e3               None              None             47    47   \n",
       "DS_0339e44248               None              None             47    47   \n",
       "DS_033b09f437       Dec 11, 2013               CGD             53    53   \n",
       "DS_033b09f437       Dec 11, 2013               CGD             51    51   \n",
       "DS_033b09f437       Dec 11, 2013               CGD             39    39   \n",
       "DS_033cf9c74d       Jun 12, 2013           GenBank             52    52   \n",
       "DS_033e62f4ec               None              None             45    45   \n",
       "DS_035232648f               None              None             47    47   \n",
       "DS_0360a7ae66       Feb 27, 2015           GenBank             56    56   \n",
       "DS_03824cf332               None              None             49    49   \n",
       "DS_03986f8069               None              None             47    47   \n",
       "DS_0399018270       Jul 29, 2015           GenBank             51    51   \n",
       "DS_0399018270                  1             Broad             21    21   \n",
       "DS_0399018270       Jul 29, 2015           GenBank             29    29   \n",
       "DS_03a864a884               None              None             47    47   \n",
       "DS_03aef8200c               None              None             48    48   \n",
       "DS_03d8e25e02               None              None             47    47   \n",
       "DS_03f94e0879               None              None             47    47   \n",
       "DS_04102dc2b8               None              None             47    47   \n",
       "DS_0445efcab5               None              None             47    47   \n",
       "DS_04464ff60b         2018-11-27           GenBank             53    53   \n",
       "DS_04666bdd43               None              None             28    28   \n",
       "\n",
       "              release_date   genome_version genome_source  \\\n",
       "dataset_id                                                  \n",
       "DS_0011425ac1   2021-03-16  GCA_000225285.2         INSDC   \n",
       "DS_0024327063   2022-02-15             None          None   \n",
       "DS_002a5dad83   2020-04-07             None          None   \n",
       "DS_00672484d6   2020-04-07             None          None   \n",
       "DS_006cb7d943   2015-07-23             None          None   \n",
       "DS_00893e0bf8   2020-04-07             None          None   \n",
       "DS_00b47a520a   2020-04-07             None          None   \n",
       "DS_011c12b8d3   2020-12-17             None          None   \n",
       "DS_01225327f4   2020-08-27             None          None   \n",
       "DS_012298da2c   2019-11-06             None          None   \n",
       "DS_019172d4ff   2020-04-07             None          None   \n",
       "DS_01b2dc5b05   2021-03-16  GCA_000149845.2       GenBank   \n",
       "DS_01b2dc5b05   2013-09-25                5         Broad   \n",
       "DS_01b2dc5b05   2016-10-12     Mar 23, 2015       GenBank   \n",
       "DS_01c6cee55d   2019-11-06     Nov 07, 2017       GenBank   \n",
       "DS_01c6cee55d   2020-11-05  GCA_000338675.2       GenBank   \n",
       "DS_01c6cee55d   2013-09-25      Feb 7, 2013       GenBank   \n",
       "DS_01d08f2525   2020-04-07             None          None   \n",
       "DS_022384d18a   2018-02-19     Nov 29, 2017        Sanger   \n",
       "DS_022384d18a   2021-03-16  GCA_900097025.1       GenBank   \n",
       "DS_0226b5025b   2020-04-07             None          None   \n",
       "DS_02354d7d63   2015-04-14     Sep 30, 2014       GenBank   \n",
       "DS_02354d7d63   2021-05-20  GCA_000760515.1       GenBank   \n",
       "DS_0241addc8e   2022-02-15  GCA_019096175.1         INSDC   \n",
       "DS_02717866d1   2020-04-07             None          None   \n",
       "DS_0273077ad5   2020-04-07             None          None   \n",
       "DS_02c0ef3cc2   2020-08-27             None          None   \n",
       "DS_02caff887d   2020-04-07             None          None   \n",
       "DS_03395fd8e3   2020-04-07             None          None   \n",
       "DS_0339e44248   2020-04-07             None          None   \n",
       "DS_033b09f437   2021-07-21  GCA_000149445.2       GenBank   \n",
       "DS_033b09f437   2021-03-16  GCA_000149445.2       GenBank   \n",
       "DS_033b09f437   2018-08-30     Dec 11, 2013           CGD   \n",
       "DS_033cf9c74d   2021-05-20  GCF_000364345.1         INSDC   \n",
       "DS_033e62f4ec   2019-09-05             None          None   \n",
       "DS_035232648f   2020-04-07             None          None   \n",
       "DS_0360a7ae66   2022-02-15  GCA_000026945.1         INSDC   \n",
       "DS_03824cf332   2020-11-05             None          None   \n",
       "DS_03986f8069   2020-04-07             None          None   \n",
       "DS_0399018270   2021-03-16  GCA_000182565.2       GenBank   \n",
       "DS_0399018270   2014-05-08                1         Broad   \n",
       "DS_0399018270   2016-10-12     Jul 29, 2015       GenBank   \n",
       "DS_03a864a884   2020-04-07             None          None   \n",
       "DS_03aef8200c   2020-08-27             None          None   \n",
       "DS_03d8e25e02   2020-04-07             None          None   \n",
       "DS_03f94e0879   2020-04-07             None          None   \n",
       "DS_04102dc2b8   2020-04-07             None          None   \n",
       "DS_0445efcab5   2020-04-07             None          None   \n",
       "DS_04464ff60b   2021-07-21  GCA_003843895.1         INSDC   \n",
       "DS_04666bdd43   2016-03-31             None          None   \n",
       "\n",
       "              functional_annotation_source functional_annotation_version  \n",
       "dataset_id                                                                \n",
       "DS_0011425ac1                         None                          None  \n",
       "DS_0024327063                         None                          None  \n",
       "DS_002a5dad83                         None                          None  \n",
       "DS_00672484d6                         None                          None  \n",
       "DS_006cb7d943                         None                          None  \n",
       "DS_00893e0bf8                         None                          None  \n",
       "DS_00b47a520a                         None                          None  \n",
       "DS_011c12b8d3                         None                          None  \n",
       "DS_01225327f4                         None                          None  \n",
       "DS_012298da2c                         None                          None  \n",
       "DS_019172d4ff                         None                          None  \n",
       "DS_01b2dc5b05                         None                          None  \n",
       "DS_01b2dc5b05                         None                          None  \n",
       "DS_01b2dc5b05                         None                          None  \n",
       "DS_01c6cee55d                         None                          None  \n",
       "DS_01c6cee55d                         None                          None  \n",
       "DS_01c6cee55d                         None                          None  \n",
       "DS_01d08f2525                         None                          None  \n",
       "DS_022384d18a                         None                          None  \n",
       "DS_022384d18a                         None                          None  \n",
       "DS_0226b5025b                         None                          None  \n",
       "DS_02354d7d63                         None                          None  \n",
       "DS_02354d7d63                         None                          None  \n",
       "DS_0241addc8e                         None                          None  \n",
       "DS_02717866d1                         None                          None  \n",
       "DS_0273077ad5                         None                          None  \n",
       "DS_02c0ef3cc2                         None                          None  \n",
       "DS_02caff887d                         None                          None  \n",
       "DS_03395fd8e3                         None                          None  \n",
       "DS_0339e44248                         None                          None  \n",
       "DS_033b09f437                         None                          None  \n",
       "DS_033b09f437                         None                          None  \n",
       "DS_033b09f437                         None                          None  \n",
       "DS_033cf9c74d                         None                          None  \n",
       "DS_033e62f4ec                         None                          None  \n",
       "DS_035232648f                         None                          None  \n",
       "DS_0360a7ae66                         None                          None  \n",
       "DS_03824cf332                         None                          None  \n",
       "DS_03986f8069                         None                          None  \n",
       "DS_0399018270                         None                          None  \n",
       "DS_0399018270                         None                          None  \n",
       "DS_0399018270                         None                          None  \n",
       "DS_03a864a884                         None                          None  \n",
       "DS_03aef8200c                         None                          None  \n",
       "DS_03d8e25e02                         None                          None  \n",
       "DS_03f94e0879                         None                          None  \n",
       "DS_04102dc2b8                         None                          None  \n",
       "DS_0445efcab5                         None                          None  \n",
       "DS_04464ff60b                         None                          None  \n",
       "DS_04666bdd43                         None                          None  "
      ]
     },
     "metadata": {},
     "execution_count": 15
    }
   ],
   "metadata": {}
  },
  {
   "cell_type": "markdown",
   "source": [
    "## `tables.Publications`"
   ],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "source": [
    "df_pubs.head()"
   ],
   "outputs": [
    {
     "output_type": "execute_result",
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pubmed_link</th>\n",
       "      <th>citation</th>\n",
       "      <th>dataset_id</th>\n",
       "      <th>pmid</th>\n",
       "      <th>url</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>http://www.ncbi.nlm.nih.gov/pubmed/23468653</td>\n",
       "      <td>Plant-symbiotic fungi as chemical engineers: multi-genome analysis of the clavicipitaceae reveals dynamics of alkaloid loci. Schardl et al. PLoS Genet 2013;9(2):e1003323</td>\n",
       "      <td>DS_0011425ac1</td>\n",
       "      <td>23468653</td>\n",
       "      <td>http://www.ncbi.nlm.nih.gov/pubmed/23468653</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>http://www.ncbi.nlm.nih.gov/pubmed/33323516</td>\n",
       "      <td>A Noncanonical DNA Damage Checkpoint Response in a Major Fungal Pathogen. Shor et al. mBio 2020 12 15;11(6):&lt;?xml version=\"1.0\" ?&gt; &lt;!DOCTYPE PubmedArticleSet PUBLIC \"-//NLM//DTD PubMedArticle, 1st January 2019//EN\" \"https://dtd.nlm.nih.gov/ncbi/pubmed/out/pubmed_190101.dtd\"&gt; &lt;PubmedArticleSet&gt;&lt;PubmedArticle&gt;</td>\n",
       "      <td>DS_0024327063</td>\n",
       "      <td>33323516</td>\n",
       "      <td>http://www.ncbi.nlm.nih.gov/pubmed/33323516</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>http://www.ncbi.nlm.nih.gov/pubmed/25266764</td>\n",
       "      <td>Proteomic analysis in Giardia duodenalis yields insights into strain virulence and antigenic variation. Emery et al. Proteomics 2014;14(21-22):2523-34</td>\n",
       "      <td>DS_006cb7d943</td>\n",
       "      <td>25266764</td>\n",
       "      <td>http://www.ncbi.nlm.nih.gov/pubmed/25266764</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>http://www.ncbi.nlm.nih.gov/pubmed/33553008</td>\n",
       "      <td>High-Resolution Mapping of Transcription Initiation in the Asexual Stages of &lt;i&gt;Toxoplasma gondii&lt;/i&gt;. Markus et al. Front Cell Infect Microbiol 2020;10(&lt;?xml version=\"1.0\" ?&gt; &lt;!DOCTYPE PubmedArticleSet PUBLIC \"-//NLM//DTD PubMedArticle, 1st January 2019//EN\" \"https://dtd.nlm.nih.gov/ncbi/pubmed/out/pubmed_190101.dtd\"&gt; &lt;PubmedArticleSet&gt; &lt;PubmedArticle&gt;):617998</td>\n",
       "      <td>DS_011c12b8d3</td>\n",
       "      <td>33553008</td>\n",
       "      <td>http://www.ncbi.nlm.nih.gov/pubmed/33553008</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>http://www.ncbi.nlm.nih.gov/pubmed/25851262</td>\n",
       "      <td>Genome sequence comparison of Aspergillus fumigatus strains isolated from patients with pulmonary aspergilloma and chronic necrotizing pulmonary aspergillosis. Takahashi-Nakaguchi et al. Med. Mycol. 2015;53(4):353-60</td>\n",
       "      <td>DS_012298da2c</td>\n",
       "      <td>25851262</td>\n",
       "      <td>http://www.ncbi.nlm.nih.gov/pubmed/25851262</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                   pubmed_link  \\\n",
       "0  http://www.ncbi.nlm.nih.gov/pubmed/23468653   \n",
       "0  http://www.ncbi.nlm.nih.gov/pubmed/33323516   \n",
       "0  http://www.ncbi.nlm.nih.gov/pubmed/25266764   \n",
       "0  http://www.ncbi.nlm.nih.gov/pubmed/33553008   \n",
       "0  http://www.ncbi.nlm.nih.gov/pubmed/25851262   \n",
       "\n",
       "                                                                                                                                                                                                                                                                                                                                                                      citation  \\\n",
       "0                                                                                                                                                                                                    Plant-symbiotic fungi as chemical engineers: multi-genome analysis of the clavicipitaceae reveals dynamics of alkaloid loci. Schardl et al. PLoS Genet 2013;9(2):e1003323   \n",
       "0                                                        A Noncanonical DNA Damage Checkpoint Response in a Major Fungal Pathogen. Shor et al. mBio 2020 12 15;11(6):<?xml version=\"1.0\" ?> <!DOCTYPE PubmedArticleSet PUBLIC \"-//NLM//DTD PubMedArticle, 1st January 2019//EN\" \"https://dtd.nlm.nih.gov/ncbi/pubmed/out/pubmed_190101.dtd\"> <PubmedArticleSet><PubmedArticle>   \n",
       "0                                                                                                                                                                                                                       Proteomic analysis in Giardia duodenalis yields insights into strain virulence and antigenic variation. Emery et al. Proteomics 2014;14(21-22):2523-34   \n",
       "0  High-Resolution Mapping of Transcription Initiation in the Asexual Stages of <i>Toxoplasma gondii</i>. Markus et al. Front Cell Infect Microbiol 2020;10(<?xml version=\"1.0\" ?> <!DOCTYPE PubmedArticleSet PUBLIC \"-//NLM//DTD PubMedArticle, 1st January 2019//EN\" \"https://dtd.nlm.nih.gov/ncbi/pubmed/out/pubmed_190101.dtd\"> <PubmedArticleSet> <PubmedArticle>):617998   \n",
       "0                                                                                                                                                     Genome sequence comparison of Aspergillus fumigatus strains isolated from patients with pulmonary aspergilloma and chronic necrotizing pulmonary aspergillosis. Takahashi-Nakaguchi et al. Med. Mycol. 2015;53(4):353-60   \n",
       "\n",
       "      dataset_id      pmid                                          url  \n",
       "0  DS_0011425ac1  23468653  http://www.ncbi.nlm.nih.gov/pubmed/23468653  \n",
       "0  DS_0024327063  33323516  http://www.ncbi.nlm.nih.gov/pubmed/33323516  \n",
       "0  DS_006cb7d943  25266764  http://www.ncbi.nlm.nih.gov/pubmed/25266764  \n",
       "0  DS_011c12b8d3  33553008  http://www.ncbi.nlm.nih.gov/pubmed/33553008  \n",
       "0  DS_012298da2c  25851262  http://www.ncbi.nlm.nih.gov/pubmed/25851262  "
      ]
     },
     "metadata": {},
     "execution_count": 6
    }
   ],
   "metadata": {}
  },
  {
   "cell_type": "markdown",
   "source": [
    "## `tables.contacts`"
   ],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "source": [
    "df_contacts.head()"
   ],
   "outputs": [
    {
     "output_type": "execute_result",
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>contact_name</th>\n",
       "      <th>affiliation</th>\n",
       "      <th>dataset_id</th>\n",
       "      <th>dataset_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Christopher Schardl</td>\n",
       "      <td>Department of Plant Pathology, University of Kentucky, USA</td>\n",
       "      <td>DS_0011425ac1</td>\n",
       "      <td>eglyE277_primary_genome_RSRC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Erika Shor</td>\n",
       "      <td>Center for Discovery and Innovation, Hackensack Meridian Health</td>\n",
       "      <td>DS_0024327063</td>\n",
       "      <td>cglaCBS138_Shor_2020_Cglabr_ebi_rnaSeq_RSRC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>David Perlin</td>\n",
       "      <td>Center for Discovery and Innovation, Hackensack Meridian Health</td>\n",
       "      <td>DS_0024327063</td>\n",
       "      <td>cglaCBS138_Shor_2020_Cglabr_ebi_rnaSeq_RSRC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Keira Lucas</td>\n",
       "      <td>Collier Mosquito Control District</td>\n",
       "      <td>DS_002a5dad83</td>\n",
       "      <td>PopBio_Study_VBP0000343_RSRC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Linda Kothera</td>\n",
       "      <td>Division of Vector-Borne Diseases, Centers for Disease Control and Prevention, Fort Collins, Colorado, United States of America</td>\n",
       "      <td>DS_00672484d6</td>\n",
       "      <td>PopBio_Study_VBP0000454_RSRC</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          contact_name  \\\n",
       "0  Christopher Schardl   \n",
       "0           Erika Shor   \n",
       "0         David Perlin   \n",
       "0          Keira Lucas   \n",
       "0        Linda Kothera   \n",
       "\n",
       "                                                                                                                       affiliation  \\\n",
       "0                                                                       Department of Plant Pathology, University of Kentucky, USA   \n",
       "0                                                                  Center for Discovery and Innovation, Hackensack Meridian Health   \n",
       "0                                                                  Center for Discovery and Innovation, Hackensack Meridian Health   \n",
       "0                                                                                                Collier Mosquito Control District   \n",
       "0  Division of Vector-Borne Diseases, Centers for Disease Control and Prevention, Fort Collins, Colorado, United States of America   \n",
       "\n",
       "      dataset_id                                 dataset_name  \n",
       "0  DS_0011425ac1                 eglyE277_primary_genome_RSRC  \n",
       "0  DS_0024327063  cglaCBS138_Shor_2020_Cglabr_ebi_rnaSeq_RSRC  \n",
       "0  DS_0024327063  cglaCBS138_Shor_2020_Cglabr_ebi_rnaSeq_RSRC  \n",
       "0  DS_002a5dad83                 PopBio_Study_VBP0000343_RSRC  \n",
       "0  DS_00672484d6                 PopBio_Study_VBP0000454_RSRC  "
      ]
     },
     "metadata": {},
     "execution_count": 7
    }
   ],
   "metadata": {}
  },
  {
   "cell_type": "markdown",
   "source": [
    "## `tables.Version`"
   ],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "source": [
    "df_version.head()"
   ],
   "outputs": [
    {
     "output_type": "execute_result",
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>organism</th>\n",
       "      <th>project_id</th>\n",
       "      <th>dataset_id</th>\n",
       "      <th>dataset_name</th>\n",
       "      <th>version</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Epichloe glyceriae E277</td>\n",
       "      <td>FungiDB</td>\n",
       "      <td>DS_0011425ac1</td>\n",
       "      <td>eglyE277_primary_genome_RSRC</td>\n",
       "      <td>2011-09-02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Candida glabrata CBS 138</td>\n",
       "      <td>FungiDB</td>\n",
       "      <td>DS_0024327063</td>\n",
       "      <td>cglaCBS138_Shor_2020_Cglabr_ebi_rnaSeq_RSRC</td>\n",
       "      <td>2020-12-15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Giardia Assemblage A isolate WB</td>\n",
       "      <td>GiardiaDB</td>\n",
       "      <td>DS_006cb7d943</td>\n",
       "      <td>gassAWB_massSpec_Emery_Virulence_Proteomics_RSRC</td>\n",
       "      <td>2014-11-12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Aedes albopictus Foshan</td>\n",
       "      <td>VectorBase</td>\n",
       "      <td>DS_00893e0bf8</td>\n",
       "      <td>aalbFoshan_SRP052205_ebi_rnaSeq_RSRC</td>\n",
       "      <td>01-2020</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Aedes aegypti LVP_AGWG</td>\n",
       "      <td>VectorBase</td>\n",
       "      <td>DS_00b47a520a</td>\n",
       "      <td>aaegLVP_AGWG_microarrayPlatform_A-MEXP-1878-Agilent_RSRC</td>\n",
       "      <td>2014-05-02</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                          organism  project_id     dataset_id  \\\n",
       "0          Epichloe glyceriae E277     FungiDB  DS_0011425ac1   \n",
       "0         Candida glabrata CBS 138     FungiDB  DS_0024327063   \n",
       "0  Giardia Assemblage A isolate WB   GiardiaDB  DS_006cb7d943   \n",
       "0          Aedes albopictus Foshan  VectorBase  DS_00893e0bf8   \n",
       "0           Aedes aegypti LVP_AGWG  VectorBase  DS_00b47a520a   \n",
       "\n",
       "                                               dataset_name     version  \n",
       "0                              eglyE277_primary_genome_RSRC  2011-09-02  \n",
       "0               cglaCBS138_Shor_2020_Cglabr_ebi_rnaSeq_RSRC  2020-12-15  \n",
       "0          gassAWB_massSpec_Emery_Virulence_Proteomics_RSRC  2014-11-12  \n",
       "0                      aalbFoshan_SRP052205_ebi_rnaSeq_RSRC     01-2020  \n",
       "0  aaegLVP_AGWG_microarrayPlatform_A-MEXP-1878-Agilent_RSRC  2014-05-02  "
      ]
     },
     "metadata": {},
     "execution_count": 8
    }
   ],
   "metadata": {}
  },
  {
   "cell_type": "markdown",
   "source": [
    "## `tables.GeneTypeCounts`"
   ],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "source": [
    "df_gtc.head()"
   ],
   "outputs": [
    {
     "output_type": "execute_result",
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_count</th>\n",
       "      <th>gene_type</th>\n",
       "      <th>dataset_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>337</td>\n",
       "      <td>ncRNA gene</td>\n",
       "      <td>DS_01b2dc5b05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>4886</td>\n",
       "      <td>protein coding gene</td>\n",
       "      <td>DS_01b2dc5b05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>9</td>\n",
       "      <td>pseudogene</td>\n",
       "      <td>DS_01b2dc5b05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>173</td>\n",
       "      <td>ncRNA gene</td>\n",
       "      <td>DS_01c6cee55d</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>9869</td>\n",
       "      <td>protein coding gene</td>\n",
       "      <td>DS_01c6cee55d</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  gene_count            gene_type     dataset_id\n",
       "0        337           ncRNA gene  DS_01b2dc5b05\n",
       "0       4886  protein coding gene  DS_01b2dc5b05\n",
       "0          9           pseudogene  DS_01b2dc5b05\n",
       "0        173           ncRNA gene  DS_01c6cee55d\n",
       "0       9869  protein coding gene  DS_01c6cee55d"
      ]
     },
     "metadata": {},
     "execution_count": 9
    }
   ],
   "metadata": {}
  },
  {
   "cell_type": "markdown",
   "source": [
    "## `tables.DatasetHistory`  "
   ],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "source": [
    "df_dh.head()"
   ],
   "outputs": [
    {
     "output_type": "execute_result",
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>note</th>\n",
       "      <th>release_number</th>\n",
       "      <th>build</th>\n",
       "      <th>release_date</th>\n",
       "      <th>dataset_id</th>\n",
       "      <th>version</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>None</td>\n",
       "      <td>51</td>\n",
       "      <td>51</td>\n",
       "      <td>2021-03-16</td>\n",
       "      <td>DS_0011425ac1</td>\n",
       "      <td>2011-09-02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>None</td>\n",
       "      <td>56</td>\n",
       "      <td>56</td>\n",
       "      <td>2022-02-15</td>\n",
       "      <td>DS_0024327063</td>\n",
       "      <td>2020-12-15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>None</td>\n",
       "      <td>47</td>\n",
       "      <td>47</td>\n",
       "      <td>2020-04-07</td>\n",
       "      <td>DS_002a5dad83</td>\n",
       "      <td>2018-10-17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>None</td>\n",
       "      <td>47</td>\n",
       "      <td>47</td>\n",
       "      <td>2020-04-07</td>\n",
       "      <td>DS_00672484d6</td>\n",
       "      <td>2019-05-22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>None</td>\n",
       "      <td>25</td>\n",
       "      <td>25</td>\n",
       "      <td>2015-07-23</td>\n",
       "      <td>DS_006cb7d943</td>\n",
       "      <td>2014-11-12</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   note release_number build release_date     dataset_id     version\n",
       "0  None             51    51   2021-03-16  DS_0011425ac1  2011-09-02\n",
       "0  None             56    56   2022-02-15  DS_0024327063  2020-12-15\n",
       "0  None             47    47   2020-04-07  DS_002a5dad83  2018-10-17\n",
       "0  None             47    47   2020-04-07  DS_00672484d6  2019-05-22\n",
       "0  None             25    25   2015-07-23  DS_006cb7d943  2014-11-12"
      ]
     },
     "metadata": {},
     "execution_count": 10
    }
   ],
   "metadata": {}
  },
  {
   "cell_type": "markdown",
   "source": [
    "## `tables.HyperLinks`"
   ],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "source": [
    "df_hl.head()"
   ],
   "outputs": [
    {
     "output_type": "execute_result",
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>hyper_link</th>\n",
       "      <th>dataset_id</th>\n",
       "      <th>text</th>\n",
       "      <th>url</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/assembly/GCA_000225285.2</td>\n",
       "      <td>DS_0011425ac1</td>\n",
       "      <td>GenBank Assembly page</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/assembly/GCA_000225285.2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/bioproject/PRJNA67247</td>\n",
       "      <td>DS_0011425ac1</td>\n",
       "      <td>NCBI Bioproject</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/bioproject/PRJNA67247</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/bioproject/PRJNA655241</td>\n",
       "      <td>DS_0024327063</td>\n",
       "      <td>NCBI BioProject</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/bioproject/PRJNA655241</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>http://proteomecentral.proteomexchange.org</td>\n",
       "      <td>DS_006cb7d943</td>\n",
       "      <td>Data located at Proteome Central under identifier PXD000452</td>\n",
       "      <td>http://proteomecentral.proteomexchange.org</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>http://www.ncbi.nlm.nih.gov/sra/?term=SRP052205</td>\n",
       "      <td>DS_00893e0bf8</td>\n",
       "      <td>SRA project id</td>\n",
       "      <td>http://www.ncbi.nlm.nih.gov/sra/?term=SRP052205</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                              hyper_link     dataset_id  \\\n",
       "0  https://www.ncbi.nlm.nih.gov/assembly/GCA_000225285.2  DS_0011425ac1   \n",
       "0     https://www.ncbi.nlm.nih.gov/bioproject/PRJNA67247  DS_0011425ac1   \n",
       "0    https://www.ncbi.nlm.nih.gov/bioproject/PRJNA655241  DS_0024327063   \n",
       "0             http://proteomecentral.proteomexchange.org  DS_006cb7d943   \n",
       "0        http://www.ncbi.nlm.nih.gov/sra/?term=SRP052205  DS_00893e0bf8   \n",
       "\n",
       "                                                          text  \\\n",
       "0                                        GenBank Assembly page   \n",
       "0                                              NCBI Bioproject   \n",
       "0                                              NCBI BioProject   \n",
       "0  Data located at Proteome Central under identifier PXD000452   \n",
       "0                                               SRA project id   \n",
       "\n",
       "                                                     url  \n",
       "0  https://www.ncbi.nlm.nih.gov/assembly/GCA_000225285.2  \n",
       "0     https://www.ncbi.nlm.nih.gov/bioproject/PRJNA67247  \n",
       "0    https://www.ncbi.nlm.nih.gov/bioproject/PRJNA655241  \n",
       "0             http://proteomecentral.proteomexchange.org  \n",
       "0        http://www.ncbi.nlm.nih.gov/sra/?term=SRP052205  "
      ]
     },
     "metadata": {},
     "execution_count": 11
    }
   ],
   "metadata": {}
  },
  {
   "cell_type": "markdown",
   "source": [
    "## `tables.TranscriptTypeCounts`  \n"
   ],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "source": [
    "df_ttc.head()"
   ],
   "outputs": [
    {
     "output_type": "execute_result",
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_type</th>\n",
       "      <th>transcript_count</th>\n",
       "      <th>dataset_id</th>\n",
       "      <th>transcript_type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ncRNA gene</td>\n",
       "      <td>15</td>\n",
       "      <td>DS_01b2dc5b05</td>\n",
       "      <td>rRNA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ncRNA gene</td>\n",
       "      <td>322</td>\n",
       "      <td>DS_01b2dc5b05</td>\n",
       "      <td>tRNA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>protein coding gene</td>\n",
       "      <td>4886</td>\n",
       "      <td>DS_01b2dc5b05</td>\n",
       "      <td>mRNA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>pseudogene</td>\n",
       "      <td>9</td>\n",
       "      <td>DS_01b2dc5b05</td>\n",
       "      <td>pseudogenic_transcript</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ncRNA gene</td>\n",
       "      <td>2</td>\n",
       "      <td>DS_01c6cee55d</td>\n",
       "      <td>rRNA</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             gene_type transcript_count     dataset_id         transcript_type\n",
       "0           ncRNA gene               15  DS_01b2dc5b05                    rRNA\n",
       "0           ncRNA gene              322  DS_01b2dc5b05                    tRNA\n",
       "0  protein coding gene             4886  DS_01b2dc5b05                    mRNA\n",
       "0           pseudogene                9  DS_01b2dc5b05  pseudogenic_transcript\n",
       "0           ncRNA gene                2  DS_01c6cee55d                    rRNA"
      ]
     },
     "metadata": {},
     "execution_count": 12
    }
   ],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "source": [],
   "outputs": [],
   "metadata": {}
  },
  {
   "cell_type": "markdown",
   "source": [
    "---"
   ],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 204,
   "source": [
    "\n",
    "# data report aggregations \n",
    "def data_report(_df):\n",
    "    print(\"\\n[INFO] data example:\")\n",
    "    display(_df.tail())\n",
    "\n",
    "    print(\"\\n[INFO] info(): \")\n",
    "    display(_df.info())\n",
    "\n",
    "    print(\"\\n[INFO] describe() \")\n",
    "    display(_df.describe().T)\n",
    "    \n",
    "    print(\"\\n[INFO] aggregations:\")\n",
    "    display(_df.aggregate(['min', 'max']))\n",
    "\n",
    "    #print(\"\\n[INFO] :\")\n",
    "    #for key in _df.columns.values:\n",
    "     # if _df[key].dtype == object:\n",
    "          #_df[key] = _df[key].astype('category')\n",
    "          #display(_df.quantile(axis=)) #axis=1 for col-wise"
   ],
   "outputs": [],
   "metadata": {}
  },
  {
   "cell_type": "markdown",
   "source": [],
   "metadata": {}
  },
  {
   "cell_type": "markdown",
   "source": [
    "## Tables"
   ],
   "metadata": {}
  },
  {
   "cell_type": "markdown",
   "source": [
    "#### `GenomeHistory` Table"
   ],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 165,
   "source": [
    "# data sample\n",
    "display(df_gh.tail(15))"
   ],
   "outputs": [
    {
     "output_type": "display_data",
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>note</th>\n",
       "      <th>annotation_source</th>\n",
       "      <th>annotation_version</th>\n",
       "      <th>release_number</th>\n",
       "      <th>genome_version</th>\n",
       "      <th>build</th>\n",
       "      <th>release_date</th>\n",
       "      <th>genome_source</th>\n",
       "      <th>dataset_id</th>\n",
       "      <th>functional_annotation_source</th>\n",
       "      <th>functional_annotation_version</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>48</td>\n",
       "      <td>None</td>\n",
       "      <td>48</td>\n",
       "      <td>2020-08-27</td>\n",
       "      <td>None</td>\n",
       "      <td>DS_ffbe968d03</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>22</td>\n",
       "      <td>None</td>\n",
       "      <td>22</td>\n",
       "      <td>2014-09-10</td>\n",
       "      <td>None</td>\n",
       "      <td>DS_ffbe968d03</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>53</td>\n",
       "      <td>None</td>\n",
       "      <td>53</td>\n",
       "      <td>2021-07-21</td>\n",
       "      <td>None</td>\n",
       "      <td>DS_ffbe968d03</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>29</td>\n",
       "      <td>None</td>\n",
       "      <td>29</td>\n",
       "      <td>2016-10-12</td>\n",
       "      <td>None</td>\n",
       "      <td>DS_ffbe968d03</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>49</td>\n",
       "      <td>None</td>\n",
       "      <td>49</td>\n",
       "      <td>2020-11-05</td>\n",
       "      <td>None</td>\n",
       "      <td>DS_ffbe968d03</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>None</td>\n",
       "      <td>GenBank</td>\n",
       "      <td>Mar 21, 2015</td>\n",
       "      <td>46</td>\n",
       "      <td>Mar 21, 2015</td>\n",
       "      <td>46</td>\n",
       "      <td>2019-11-06</td>\n",
       "      <td>GenBank</td>\n",
       "      <td>DS_ffbe968d03</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rebuild with INSDC accession added</td>\n",
       "      <td>GenBank</td>\n",
       "      <td>Mar 21, 2015</td>\n",
       "      <td>51</td>\n",
       "      <td>GCA_000365145.2</td>\n",
       "      <td>51</td>\n",
       "      <td>2021-03-16</td>\n",
       "      <td>GenBank</td>\n",
       "      <td>DS_ffbe968d03</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>47</td>\n",
       "      <td>None</td>\n",
       "      <td>47</td>\n",
       "      <td>2020-04-07</td>\n",
       "      <td>None</td>\n",
       "      <td>DS_ffbe968d03</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>47</td>\n",
       "      <td>None</td>\n",
       "      <td>47</td>\n",
       "      <td>2020-04-07</td>\n",
       "      <td>None</td>\n",
       "      <td>DS_ffbe968d03</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>28</td>\n",
       "      <td>None</td>\n",
       "      <td>28</td>\n",
       "      <td>2016-03-31</td>\n",
       "      <td>None</td>\n",
       "      <td>DS_ffbe968d03</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>53</td>\n",
       "      <td>None</td>\n",
       "      <td>53</td>\n",
       "      <td>2021-07-21</td>\n",
       "      <td>None</td>\n",
       "      <td>DS_ffbe968d03</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>47</td>\n",
       "      <td>None</td>\n",
       "      <td>47</td>\n",
       "      <td>2020-04-07</td>\n",
       "      <td>None</td>\n",
       "      <td>DS_ffbe968d03</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>34</td>\n",
       "      <td>None</td>\n",
       "      <td>34</td>\n",
       "      <td>2017-09-07</td>\n",
       "      <td>None</td>\n",
       "      <td>DS_ffbe968d03</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>38</td>\n",
       "      <td>None</td>\n",
       "      <td>38</td>\n",
       "      <td>2018-07-05</td>\n",
       "      <td>None</td>\n",
       "      <td>DS_ffbe968d03</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>None</td>\n",
       "      <td>GenBank</td>\n",
       "      <td>Feb 02, 2017</td>\n",
       "      <td>52</td>\n",
       "      <td>GCA_900119595.1</td>\n",
       "      <td>52</td>\n",
       "      <td>2021-05-20</td>\n",
       "      <td>INSDC</td>\n",
       "      <td>DS_ffbe968d03</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                 note annotation_source annotation_version  \\\n",
       "0                                None              None               None   \n",
       "0                                None              None               None   \n",
       "0                                None              None               None   \n",
       "0                                None              None               None   \n",
       "0                                None              None               None   \n",
       "0                                None           GenBank       Mar 21, 2015   \n",
       "0  rebuild with INSDC accession added           GenBank       Mar 21, 2015   \n",
       "0                                None              None               None   \n",
       "0                                None              None               None   \n",
       "0                                None              None               None   \n",
       "0                                None              None               None   \n",
       "0                                None              None               None   \n",
       "0                                None              None               None   \n",
       "0                                None              None               None   \n",
       "0                                None           GenBank       Feb 02, 2017   \n",
       "\n",
       "  release_number   genome_version build release_date genome_source  \\\n",
       "0             48             None    48   2020-08-27          None   \n",
       "0             22             None    22   2014-09-10          None   \n",
       "0             53             None    53   2021-07-21          None   \n",
       "0             29             None    29   2016-10-12          None   \n",
       "0             49             None    49   2020-11-05          None   \n",
       "0             46     Mar 21, 2015    46   2019-11-06       GenBank   \n",
       "0             51  GCA_000365145.2    51   2021-03-16       GenBank   \n",
       "0             47             None    47   2020-04-07          None   \n",
       "0             47             None    47   2020-04-07          None   \n",
       "0             28             None    28   2016-03-31          None   \n",
       "0             53             None    53   2021-07-21          None   \n",
       "0             47             None    47   2020-04-07          None   \n",
       "0             34             None    34   2017-09-07          None   \n",
       "0             38             None    38   2018-07-05          None   \n",
       "0             52  GCA_900119595.1    52   2021-05-20         INSDC   \n",
       "\n",
       "      dataset_id functional_annotation_source functional_annotation_version  \n",
       "0  DS_ffbe968d03                         None                          None  \n",
       "0  DS_ffbe968d03                         None                          None  \n",
       "0  DS_ffbe968d03                         None                          None  \n",
       "0  DS_ffbe968d03                         None                          None  \n",
       "0  DS_ffbe968d03                         None                          None  \n",
       "0  DS_ffbe968d03                         None                          None  \n",
       "0  DS_ffbe968d03                         None                          None  \n",
       "0  DS_ffbe968d03                         None                          None  \n",
       "0  DS_ffbe968d03                         None                          None  \n",
       "0  DS_ffbe968d03                         None                          None  \n",
       "0  DS_ffbe968d03                         None                          None  \n",
       "0  DS_ffbe968d03                         None                          None  \n",
       "0  DS_ffbe968d03                         None                          None  \n",
       "0  DS_ffbe968d03                         None                          None  \n",
       "0  DS_ffbe968d03                         None                          None  "
      ]
     },
     "metadata": {}
    }
   ],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 123,
   "source": [
    "df_gh.shape"
   ],
   "outputs": [
    {
     "output_type": "execute_result",
     "data": {
      "text/plain": [
       "(3377, 11)"
      ]
     },
     "metadata": {},
     "execution_count": 123
    }
   ],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 124,
   "source": [
    "# quick info stats \n",
    "display(df_gh.info())"
   ],
   "outputs": [
    {
     "output_type": "stream",
     "name": "stdout",
     "text": [
      "<class 'pandas.core.frame.DataFrame'>\n",
      "Int64Index: 3377 entries, 0 to 0\n",
      "Data columns (total 11 columns):\n",
      " #   Column                         Non-Null Count  Dtype \n",
      "---  ------                         --------------  ----- \n",
      " 0   note                           960 non-null    object\n",
      " 1   annotation_version             1513 non-null   object\n",
      " 2   annotation_source              1513 non-null   object\n",
      " 3   release_number                 3377 non-null   object\n",
      " 4   build                          3377 non-null   object\n",
      " 5   release_date                   3377 non-null   object\n",
      " 6   genome_version                 1558 non-null   object\n",
      " 7   genome_source                  1558 non-null   object\n",
      " 8   dataset_id                     3377 non-null   object\n",
      " 9   functional_annotation_source   325 non-null    object\n",
      " 10  functional_annotation_version  325 non-null    object\n",
      "dtypes: object(11)\n",
      "memory usage: 316.6+ KB\n"
     ]
    },
    {
     "output_type": "display_data",
     "data": {
      "text/plain": [
       "None"
      ]
     },
     "metadata": {}
    }
   ],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 158,
   "source": [
    "# describe data\n",
    "display(df_gh.describe())"
   ],
   "outputs": [
    {
     "output_type": "display_data",
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>note</th>\n",
       "      <th>annotation_source</th>\n",
       "      <th>annotation_version</th>\n",
       "      <th>release_number</th>\n",
       "      <th>genome_version</th>\n",
       "      <th>build</th>\n",
       "      <th>release_date</th>\n",
       "      <th>genome_source</th>\n",
       "      <th>dataset_id</th>\n",
       "      <th>functional_annotation_source</th>\n",
       "      <th>functional_annotation_version</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>count</th>\n",
       "      <td>960</td>\n",
       "      <td>1513</td>\n",
       "      <td>1513</td>\n",
       "      <td>3377</td>\n",
       "      <td>1558</td>\n",
       "      <td>3377</td>\n",
       "      <td>3377</td>\n",
       "      <td>1558</td>\n",
       "      <td>3377</td>\n",
       "      <td>325</td>\n",
       "      <td>325</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>unique</th>\n",
       "      <td>112</td>\n",
       "      <td>61</td>\n",
       "      <td>482</td>\n",
       "      <td>45</td>\n",
       "      <td>950</td>\n",
       "      <td>45</td>\n",
       "      <td>45</td>\n",
       "      <td>53</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>35</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>top</th>\n",
       "      <td>rebuild with INSDC accession added</td>\n",
       "      <td>GenBank</td>\n",
       "      <td>n/a</td>\n",
       "      <td>47</td>\n",
       "      <td>May 28, 2016</td>\n",
       "      <td>47</td>\n",
       "      <td>2020-04-07</td>\n",
       "      <td>GenBank</td>\n",
       "      <td>DS_ffbe968d03</td>\n",
       "      <td>GeneDB</td>\n",
       "      <td>May 06, 2020</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>freq</th>\n",
       "      <td>276</td>\n",
       "      <td>638</td>\n",
       "      <td>128</td>\n",
       "      <td>1096</td>\n",
       "      <td>42</td>\n",
       "      <td>1096</td>\n",
       "      <td>1096</td>\n",
       "      <td>814</td>\n",
       "      <td>3377</td>\n",
       "      <td>264</td>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                      note annotation_source  \\\n",
       "count                                  960              1513   \n",
       "unique                                 112                61   \n",
       "top     rebuild with INSDC accession added           GenBank   \n",
       "freq                                   276               638   \n",
       "\n",
       "       annotation_version release_number genome_version build release_date  \\\n",
       "count                1513           3377           1558  3377         3377   \n",
       "unique                482             45            950    45           45   \n",
       "top                   n/a             47   May 28, 2016    47   2020-04-07   \n",
       "freq                  128           1096             42  1096         1096   \n",
       "\n",
       "       genome_source     dataset_id functional_annotation_source  \\\n",
       "count           1558           3377                          325   \n",
       "unique            53              1                            3   \n",
       "top          GenBank  DS_ffbe968d03                       GeneDB   \n",
       "freq             814           3377                          264   \n",
       "\n",
       "       functional_annotation_version  \n",
       "count                            325  \n",
       "unique                            35  \n",
       "top                     May 06, 2020  \n",
       "freq                              30  "
      ]
     },
     "metadata": {}
    }
   ],
   "metadata": {}
  },
  {
   "cell_type": "markdown",
   "source": [
    "\n",
    "\n",
    "### `records.tables.GenomeHistory`   \n",
    "  \n",
    " \"Is this a reference genome that this dataset is based on? If so, I recommend capturing with `isBasedOn`\"\n",
    "- No, looking closer it seems to be developer notes, and only has 960 non-null values, (out of 3377 records)"
   ],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 136,
   "source": [
    "# inspect notes \n",
    "mask=df_gh.note.notnull()\n",
    "note_series=df_gh.note.where(mask).dropna()"
   ],
   "outputs": [],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 137,
   "source": [
    "display(note_series[:10])"
   ],
   "outputs": [
    {
     "output_type": "display_data",
     "data": {
      "text/plain": [
       "0                   rebuild with INSDC accession added\n",
       "0                    re-annotated, data source changed\n",
       "0                                      newly-annotated\n",
       "0    consistent date format and INSDC accession whe...\n",
       "0                   rebuild with INSDC accession added\n",
       "0                   rebuild with INSDC accession added\n",
       "0               functional annotation updated from CGD\n",
       "0                   rebuild with INSDC accession added\n",
       "0                   rebuild with INSDC accession added\n",
       "0                    re-annotated, data source changed\n",
       "Name: note, dtype: object"
      ]
     },
     "metadata": {}
    }
   ],
   "metadata": {}
  },
  {
   "cell_type": "markdown",
   "source": [
    "`records.tables.GenomeHistory.annotation_source`   \n",
    "`records.tables.GenomeHistory.annotation_version`  "
   ],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 161,
   "source": [
    "mask=df_gh[['annotation_source', 'annotation_version']].notnull()\n",
    "annot_df=df_gh[['annotation_source', 'annotation_version']].where(mask).dropna()\n",
    "annot_df.tail()"
   ],
   "outputs": [
    {
     "output_type": "execute_result",
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>annotation_source</th>\n",
       "      <th>annotation_version</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>SchnauferLab</td>\n",
       "      <td>Jun 03, 2014</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>SchnauferLab</td>\n",
       "      <td>Jun 13, 2013</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GenBank</td>\n",
       "      <td>Mar 21, 2015</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GenBank</td>\n",
       "      <td>Mar 21, 2015</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GenBank</td>\n",
       "      <td>Feb 02, 2017</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  annotation_source annotation_version\n",
       "0      SchnauferLab       Jun 03, 2014\n",
       "0      SchnauferLab       Jun 13, 2013\n",
       "0           GenBank       Mar 21, 2015\n",
       "0           GenBank       Mar 21, 2015\n",
       "0           GenBank       Feb 02, 2017"
      ]
     },
     "metadata": {},
     "execution_count": 161
    }
   ],
   "metadata": {}
  },
  {
   "cell_type": "markdown",
   "source": [
    "`records.tables.GenomeHistory.release_number`    \n",
    "`records.tables.GenomeHistory.release_date`"
   ],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 164,
   "source": [
    "display(df_gh.filter(regex=\"release\").tail())\n"
   ],
   "outputs": [
    {
     "output_type": "display_data",
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>release_number</th>\n",
       "      <th>release_date</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>53</td>\n",
       "      <td>2021-07-21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>47</td>\n",
       "      <td>2020-04-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>34</td>\n",
       "      <td>2017-09-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>38</td>\n",
       "      <td>2018-07-05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>52</td>\n",
       "      <td>2021-05-20</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  release_number release_date\n",
       "0             53   2021-07-21\n",
       "0             47   2020-04-07\n",
       "0             34   2017-09-07\n",
       "0             38   2018-07-05\n",
       "0             52   2021-05-20"
      ]
     },
     "metadata": {}
    }
   ],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 144,
   "source": [
    "# versions  -- annotation_version, genome_version, functional_annotation_version\n",
    "version_df= df_gh.filter(regex=\"version\")\n",
    "vr_mask=version_df.notnull()\n",
    "vr_df=version_df.where(mask).dropna()\n",
    "\n",
    "print(\"\\n[INFO] number of unique values: \")\n",
    "display(vr_df.nunique())\n",
    "\n",
    "print(\"\\n[INFO] total non-null values: \")\n",
    "display(vr_df.count())\n"
   ],
   "outputs": [
    {
     "output_type": "stream",
     "name": "stdout",
     "text": [
      "\n",
      "[INFO] number of unique values: \n"
     ]
    },
    {
     "output_type": "display_data",
     "data": {
      "text/plain": [
       "annotation_version               36\n",
       "genome_version                   70\n",
       "functional_annotation_version    35\n",
       "dtype: int64"
      ]
     },
     "metadata": {}
    },
    {
     "output_type": "stream",
     "name": "stdout",
     "text": [
      "\n",
      "[INFO] total non-null values: \n"
     ]
    },
    {
     "output_type": "display_data",
     "data": {
      "text/plain": [
       "annotation_version               323\n",
       "genome_version                   323\n",
       "functional_annotation_version    323\n",
       "dtype: int64"
      ]
     },
     "metadata": {}
    }
   ],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 187,
   "source": [
    "vr_df.head(15)"
   ],
   "outputs": [
    {
     "output_type": "execute_result",
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>annotation_version</th>\n",
       "      <th>genome_version</th>\n",
       "      <th>functional_annotation_version</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Sep 27, 2015</td>\n",
       "      <td>GCA_000002655.1</td>\n",
       "      <td>Sep 01, 2020</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Sep 27, 2015</td>\n",
       "      <td>Sep 27, 2015</td>\n",
       "      <td>May 04, 2018</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Sep 27, 2015</td>\n",
       "      <td>GCA_000002655.1</td>\n",
       "      <td>Oct 01, 2021</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Sep 27, 2015</td>\n",
       "      <td>Sep 27, 2015</td>\n",
       "      <td>Nov 21, 2017</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Sep 27, 2015</td>\n",
       "      <td>Sep 27, 2015</td>\n",
       "      <td>May 20, 2017</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Sep 27, 2015</td>\n",
       "      <td>Sep 27, 2015</td>\n",
       "      <td>Sep 01, 2020</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Sep 27, 2015</td>\n",
       "      <td>Sep 27, 2015</td>\n",
       "      <td>May 06, 2020</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Sep 27, 2015</td>\n",
       "      <td>Sep 27, 2015</td>\n",
       "      <td>Jul 26, 2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Sep 27, 2015</td>\n",
       "      <td>Sep 27, 2015</td>\n",
       "      <td>Jan 15, 2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Sep 27, 2015</td>\n",
       "      <td>Sep 27, 2015</td>\n",
       "      <td>Feb 16, 2017</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Oct 04, 2015</td>\n",
       "      <td>Oct 04, 2015</td>\n",
       "      <td>Jan 15, 2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Oct 04, 2015</td>\n",
       "      <td>GCA_000184455.3</td>\n",
       "      <td>Jan 15, 2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Dec 07, 2015</td>\n",
       "      <td>Dec 07, 2015</td>\n",
       "      <td>Oct 22, 2018</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Dec 07, 2015</td>\n",
       "      <td>Dec 07, 2015</td>\n",
       "      <td>Jan 15, 2019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Dec 07, 2015</td>\n",
       "      <td>Dec 07, 2015</td>\n",
       "      <td>Mar 30, 2019</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  annotation_version   genome_version functional_annotation_version\n",
       "0       Sep 27, 2015  GCA_000002655.1                  Sep 01, 2020\n",
       "0       Sep 27, 2015     Sep 27, 2015                  May 04, 2018\n",
       "0       Sep 27, 2015  GCA_000002655.1                  Oct 01, 2021\n",
       "0       Sep 27, 2015     Sep 27, 2015                  Nov 21, 2017\n",
       "0       Sep 27, 2015     Sep 27, 2015                  May 20, 2017\n",
       "0       Sep 27, 2015     Sep 27, 2015                  Sep 01, 2020\n",
       "0       Sep 27, 2015     Sep 27, 2015                  May 06, 2020\n",
       "0       Sep 27, 2015     Sep 27, 2015                  Jul 26, 2019\n",
       "0       Sep 27, 2015     Sep 27, 2015                  Jan 15, 2019\n",
       "0       Sep 27, 2015     Sep 27, 2015                  Feb 16, 2017\n",
       "0       Oct 04, 2015     Oct 04, 2015                  Jan 15, 2019\n",
       "0       Oct 04, 2015  GCA_000184455.3                  Jan 15, 2019\n",
       "0       Dec 07, 2015     Dec 07, 2015                  Oct 22, 2018\n",
       "0       Dec 07, 2015     Dec 07, 2015                  Jan 15, 2019\n",
       "0       Dec 07, 2015     Dec 07, 2015                  Mar 30, 2019"
      ]
     },
     "metadata": {},
     "execution_count": 187
    }
   ],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 166,
   "source": [
    "# non-null large dataset\n",
    "nonnull_mask=df_gh.notnull()\n",
    "nonnull_df=df_gh.where(nonnull_mask).dropna()"
   ],
   "outputs": [],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 172,
   "source": [
    "nonnull_df['annotation_source'].unique(), nonnull_df['genome_source'].unique(), nonnull_df['functional_annotation_source'].unique()"
   ],
   "outputs": [
    {
     "output_type": "execute_result",
     "data": {
      "text/plain": [
       "(array(['AspGD', 'GeneDB', 'n/a', 'GenBank', 'RequenaLab', 'Broad',\n",
       "        'HammondKosackLab', 'Sanger'], dtype=object),\n",
       " array(['GenBank', 'AspGD', 'GeneDB', 'RequenaLab', 'Broad',\n",
       "        'HammondKosackLab'], dtype=object),\n",
       " array(['Sanger', 'GeneDB', 'GenBank'], dtype=object))"
      ]
     },
     "metadata": {},
     "execution_count": 172
    }
   ],
   "metadata": {}
  },
  {
   "cell_type": "code",
   "execution_count": 184,
   "source": [
    "df_gh.loc[df_gh['dataset_id'] == \"DS_ffbe968d03\"].head()\n"
   ],
   "outputs": [
    {
     "output_type": "execute_result",
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>note</th>\n",
       "      <th>annotation_source</th>\n",
       "      <th>annotation_version</th>\n",
       "      <th>release_number</th>\n",
       "      <th>genome_version</th>\n",
       "      <th>build</th>\n",
       "      <th>release_date</th>\n",
       "      <th>genome_source</th>\n",
       "      <th>dataset_id</th>\n",
       "      <th>functional_annotation_source</th>\n",
       "      <th>functional_annotation_version</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>51</td>\n",
       "      <td>GCA_000225285.2</td>\n",
       "      <td>51</td>\n",
       "      <td>2021-03-16</td>\n",
       "      <td>INSDC</td>\n",
       "      <td>DS_ffbe968d03</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>56</td>\n",
       "      <td>None</td>\n",
       "      <td>56</td>\n",
       "      <td>2022-02-15</td>\n",
       "      <td>None</td>\n",
       "      <td>DS_ffbe968d03</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>47</td>\n",
       "      <td>None</td>\n",
       "      <td>47</td>\n",
       "      <td>2020-04-07</td>\n",
       "      <td>None</td>\n",
       "      <td>DS_ffbe968d03</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>47</td>\n",
       "      <td>None</td>\n",
       "      <td>47</td>\n",
       "      <td>2020-04-07</td>\n",
       "      <td>None</td>\n",
       "      <td>DS_ffbe968d03</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>25</td>\n",
       "      <td>None</td>\n",
       "      <td>25</td>\n",
       "      <td>2015-07-23</td>\n",
       "      <td>None</td>\n",
       "      <td>DS_ffbe968d03</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   note annotation_source annotation_version release_number   genome_version  \\\n",
       "0  None              None               None             51  GCA_000225285.2   \n",
       "0  None              None               None             56             None   \n",
       "0  None              None               None             47             None   \n",
       "0  None              None               None             47             None   \n",
       "0  None              None               None             25             None   \n",
       "\n",
       "  build release_date genome_source     dataset_id  \\\n",
       "0    51   2021-03-16         INSDC  DS_ffbe968d03   \n",
       "0    56   2022-02-15          None  DS_ffbe968d03   \n",
       "0    47   2020-04-07          None  DS_ffbe968d03   \n",
       "0    47   2020-04-07          None  DS_ffbe968d03   \n",
       "0    25   2015-07-23          None  DS_ffbe968d03   \n",
       "\n",
       "  functional_annotation_source functional_annotation_version  \n",
       "0                         None                          None  \n",
       "0                         None                          None  \n",
       "0                         None                          None  \n",
       "0                         None                          None  \n",
       "0                         None                          None  "
      ]
     },
     "metadata": {},
     "execution_count": 184
    }
   ],
   "metadata": {}
  },
  {
   "cell_type": "markdown",
   "source": [
    "---"
   ],
   "metadata": {}
  }
 ],
 "metadata": {
  "orig_nbformat": 4,
  "language_info": {
   "name": "python",
   "version": "3.8.11",
   "mimetype": "text/x-python",
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "pygments_lexer": "ipython3",
   "nbconvert_exporter": "python",
   "file_extension": ".py"
  },
  "kernelspec": {
   "name": "python3",
   "display_name": "Python 3.8.11 64-bit ('su_lab': conda)"
  },
  "interpreter": {
   "hash": "ef05315d5f008dd046167e10c6ddffaed5c65a031bde4d25d7d1e2bf86440ab1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}